assets/build/install.sh

```diff
@@ -2,47 +2,49 @@
 
 set -e
 
-source ${SALT_BUILD_DIR}/functions.sh
+source "${SALT_BUILD_DIR}/functions.sh"
 
 # Install build dependencies
 echo "Installing dependencies ..."
-BUILD_DEPENDENCIES="cmake gcc g++ make \
-    libhttp-parser-dev libssl-dev zlib1g-dev \
-    libcurl4-openssl-dev libffi-dev swig"
+BUILD_DEPENDENCIES=(
+  cmake gcc g++ make \
+  libhttp-parser-dev libssl-dev zlib1g-dev \
+  libcurl4-openssl-dev libffi-dev swig \
+)
 
 apt-get update
-DEBIAN_FRONTEND=noninteractive apt-get install --yes --quiet --no-install-recommends ${BUILD_DEPENDENCIES}
+DEBIAN_FRONTEND=noninteractive apt-get install --yes --quiet --no-install-recommends "${BUILD_DEPENDENCIES[@]}"
 
 # Create salt user
 echo "Creating ${SALT_USER} user ..."
-useradd -d ${SALT_HOME} -ms /bin/bash -U -G root,sudo ${SALT_USER}
+useradd -d "${SALT_HOME}" -ms /bin/bash -U -G root,sudo,shadow "${SALT_USER}"
```

Comment:
Salt user must belong to shadow group in order to read /etc/shadow for salt-api service

https://github.com/saltstack/salt/issues/7762